lib/elixir/lib/kernel/lexical_tracker.ex

```diff
@@ -98,61 +116,79 @@ defmodule Kernel.LexicalTracker do
 
   # Callbacks
 
-  def init(dest) do
-    {:ok, {:ets.new(__MODULE__, [:protected]), dest}}
+  def init({dest, parent}) do
+    {:ok, {:ets.new(__MODULE__, [:protected]), dest, parent}}
   end
 
   @doc false
-  def handle_call(:ets, _from, {d, dest}) do
-    {:reply, d, {d, dest}}
+  def handle_call(:ets, _from, {d, dest, parent}) do
+    {:reply, d, {d, dest, parent}}
   end
 
-  def handle_call(:dest, _from, {d, dest}) do
-    {:reply, dest, {d, dest}}
+  def handle_call(:dest, _from, {d, dest, parent}) do
+    {:reply, dest, {d, dest, parent}}
   end
 
-  def handle_cast({:remote_dispatch, module, mode}, {d, dest}) do
+  def handle_cast({:remote_dispatch, module, mode}, {d, dest, parent}) do
+    if parent, do: remote_dispatch(parent, module, mode)
+
     add_reference(d, module, mode)
-    {:noreply, {d, dest}}
+    {:noreply, {d, dest, parent}}
   end
 
-  def handle_cast({:import_dispatch, {module, function, arity}}, {d, dest}) do
+  def handle_cast({:remote_dispatch, module, fas, line, mode}, {d, dest, parent}) do
+    if parent, do: remote_dispatch(parent, module, fas, line, mode)
+
+    add_reference(d, module, mode)
+    add_reference(d, module, fas, line, mode)
+    {:noreply, {d, dest, parent}}
+  end
+
+  def handle_cast({:import_dispatch, {module, function, arity}}, {d, dest, parent}) do
```

Comment:
I tried adding the line to `:import_dispatch` as you suggested, @josevalim, but I couldn't really figure out how it should be used. 

It seems dispatches to imported funcs end up making it through `:remote_dispatch`, so they show up in the call graph, and it didn't seem semantically correct to add them to the "compile_remote_calls" from here (which is all I could figure might make sense), because they aren't really called at compile time. The current functionality involving import warnings didn't seem to need the line data either, so I just decided to leave it out.

If I should add it back in, can you describe where the data would be used and how?